### DISCUSSION TITLE
Optimisation in deployment

### DISCUSSION THREAD

**Discussion by SSROCKS30** [pallets/flask#5520] [Q&A] [answered]
I'm seeking optimization strategies for a Flask-based face recognition app using YOLO and DeepFace libraries. The app works fine on localhost but consistently hits memory limits on free-tier hosting services. What are the best practices to reduce memory usage in Flask apps with ML models? Looking for specific optimizations for resource-constrained platforms.

**Comment by ThiefMaster** [comment]
Not using free hosting for ML and other compute-heavy deployment would probably be a good idea.

Also, this sounds pretty much like a case on how to optimize your ML usage, and not how to optimize Flask (IMHO those two things should even run separately and not in the same process)

**Comment by NikhilKalloli** [comment, marked as answer]
Consider using [Streamlit](https://streamlit.io/cloud) instead of Flask. It offers better memory allocation for ML projects like your face recognition app. This can help address your deployment memory issues. For implementation details, check [this](https://github.com/NikhilKalloli/Face-Recognition)
